crates/pallet-domains/src/lib.rs

```diff
@@ -1476,6 +1487,44 @@ impl<T: Config> Pallet<T> {
             FraudProofError::ChallengingGenesisReceipt
         );
 
+        if let FraudProof::InvalidBundle(InvalidBundleProof::TotalRewards(
+            InvalidTotalRewardsProof {
+                domain_id,
+                block_number,
+                domain_block_hash,
+                storage_proof,
+                ..
+            },
+        )) = fraud_proof
+        {
+            let domain_number = T::DomainNumber::decode(&mut block_number.encode().as_slice())
+                .map_err(|_| FraudProofError::FailedToDecodeDomainBlockNumber)?;
+            let domain_block_hash =
+                T::DomainHash::decode(&mut domain_block_hash.encode().as_slice())
+                    .map_err(|_| FraudProofError::FailedToDecodeDomainBlockHash)?;
+            let state_root = T::Hash::decode(
+                &mut StateRoots::<T>::get((domain_id, domain_number, domain_block_hash))
+                    .ok_or(FraudProofError::MissingDomainStateRoot)?
+                    .encode()
+                    .as_slice(),
+            )
+            .map_err(|_| FraudProofError::FailedToDecodeDomainBlockHash)?;
```

Comment:
We should use the `bad_receipt.final_state_root` instead because:
- The invalid reward info fraud proof is dependent on the invalid state transition fraud proof thus `final_state_root` is trust worth
- The state roots in `StateRoots` are submitted with ER, if it is from another fraudulent ER with invalid `final_state_root` then the verification may pass for an invalid fraud proof.